From 917221bb1513f974c048ddb545bef1e6475f23fd Mon Sep 17 00:00:00 2001
From: Olivier Huber <oli.huber@gmail.com>
Date: Wed, 30 Mar 2022 22:59:08 +0200
Subject: [PATCH 1/2] Use CMAKE_INSTALL_LIBDIR to specify LIBINSTDIR on Linux

The directory for system libraries is a nice conundrum on Linux.
It could be lib64 or lib or lib/<multiarch-tuple>. To sort this out,
CMake provides via GNUInstallDirs the CMAKE_INSTALL_LIBDIR variable,
see https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html
The distributions are then free to set this according to their tastes.
---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 40a669d6a..625d87ab8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,7 +94,8 @@ ELSE()
   IF (WIN32)
     SET(LIBINSTDIR "bin")
   ELSE ()
-    SET(LIBINSTDIR "lib" CACHE STRING "library directory")
+    include(GNUInstallDirs)
+    SET(LIBINSTDIR "${CMAKE_INSTALL_LIBDIR}" CACHE STRING "library directory")
   ENDIF ()
   SET(RESOURCEINSTDIR "share/${PROJECT_NAME}")
   SET(RPATH_PREFIX "$ORIGIN")

From 2fae4f33dcbd7023a587b045ae0e0d7129c65866 Mon Sep 17 00:00:00 2001
From: Olivier Huber <oli.huber@gmail.com>
Date: Wed, 24 Aug 2022 21:59:27 +0200
Subject: [PATCH 2/2] Document GNUInstallDirs

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 625d87ab8..45c962ec5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,6 +94,8 @@ ELSE()
   IF (WIN32)
     SET(LIBINSTDIR "bin")
   ELSE ()
+    # See https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html for
+    # the purpose of this module
     include(GNUInstallDirs)
     SET(LIBINSTDIR "${CMAKE_INSTALL_LIBDIR}" CACHE STRING "library directory")
   ENDIF ()
